internal/services/pubkey_service.go

```diff
@@ -94,6 +98,29 @@ func DeletePubkey(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	pubkeyResource, err := dao.GetPubkeyResourceDao(r.Context())
+	if err != nil {
+		renderError(w, r, payloads.NewDAOError(r.Context(), "delete pubkey", err))
+		return
+	}
+
+	resources, err := pubkeyResource.ListByPubkeyId(r.Context(), id)
+	if err != nil {
+		renderError(w, r, payloads.NewDAOError(r.Context(), "delete pubkey", err))
+		return
+	}
+
+	for _, res := range resources {
+		if res.Handle != "" {
+			err = ec2Client.DeleteSSHKey(res.Handle)
```

Comment:
This will not work IMHO, we need to get ARN and assume role and get the sts client, because ec2 deals with our account, but the key is in the customer account.

IMHO we might also want to log with Info lvl, that we are deleting that key :thinking:
It's an action on the customer account, so there should be a trace for such action.

Reply:
I guess we should implement a check - if any operation other than assume role is executed on the service account, error out immediately. Just an idea...